docs/60_Usage_Best_Practices.md

```diff
@@ -0,0 +1,302 @@
+
+This indicates that the `my_field` field must be present in each document within the `my_single_view` collection, and it must have a unique value across all documents in the collection. The unique index ensures the uniqueness constraint on the `my_field` field, preventing duplicate values from being inserted into the collection.
+
+## Pagination
+
+A possible implementation to read data from the collection is to utilize client-side pagination by invoking the GET method with the query parameters `_sk` (skip) and `_l` (limit):
+
+The `_sk` parameter is used to specify the number of records to skip before starting to retrieve the result set. It allows you to navigate through different pages of data;
+
+The `_l` parameter represents the maximum number of records to retrieve in a single page. It determines the size of each page of data.
+
+By adjusting the values of `_sk` and `_l`, you can control the pagination of data retrieval based on your specific requirements. For example, to retrieve the first page of data, you would set `_sk=0` and `_l` to the desired page size. For subsequent pages, you would increase the value of `_sk` by the page size to skip the already retrieved records.
+
+:::info
+If the `_sk` and `_l` parameters are not provided to the CRUD Service, the default behavior would be to apply a zero offset to the MongoDB cursor with a size equal to the value specified in the `CRUD_MAX_LIMIT` environment variable. The default value for `CRUD_MAX_LIMIT` is set to 200.
+:::
+
+Let's assume we have a collection consisting of 100 records, and we need to implement a pagination mechanism to retrieve 25 records at a time until all records are fetched.
+
+This procedure translates into the following 4 HTTP requests to the GET /my_single_view/ endpoint:
+
+- GET /my_single_view/?_sk=0&_l=25
+- GET /my_single_view/?_sk=25&_l=25
+- GET /my_single_view/?_sk=50&_l=25
+- GET /my_single_view/?_sk=75&_l=25
+
+Now, we want to filter the records based on the `updatedAt` field, which is updated by other services (such as the Single View Creator) to indicate the time when the record was last modified. 
+
+Specifically, we want to try retrieving paginated records updated within the last 10 minutes.
+
+The previously written procedure can be adapted as follows:
+
+- GET /my_single_view/?_sk=0&_l=25&_q=%7B%22updatedAt%22:%7B%22$gte%22:%22%22%7D%7D
+- GET /my_single_view/?_sk=25&_l=25&_q=%7B%22updatedAt%22:%7B%22$gte%22:%22your%20iso%20date%22%7D%7D
+- GET /my_single_view/?_sk=50&_l=25&_q=%7B%22updatedAt%22:%7B%22$gte%22:%22your%20iso%20date%22%7D%7D
+- GET /my_single_view/?_sk=75&_l=25&_q=%7B%22updatedAt%22:%7B%22$gte%22:%22your%20iso%20date%22%7D%7D
+
+In the above requests, the filter defined translates to the following MongoDB query contained in the `_q` query parameter:
+
+```json
+{"updatedAt":{"$gte":"your iso date"}}
+```
```

Comment:
I was thinking if it makes sense to put this before the GET bullet point.

The request contains a query paramter really difficult to read (`&_q=%7B%22updatedAt%22:%7B%22$gte%22:%22your%20iso%20date%22%7D%7D`), and I think that explaining our case before to show examples can improve the readibility of the page.